rotkehlchen/api/rest.py

```diff
@@ -3663,9 +3663,13 @@ def get_report_data(self, filter_query: ReportDataFilterQuery) -> Response:
         if self.rotkehlchen.premium is None:
             with_limit = True
             entries_limit = FREE_PNL_EVENTS_LIMIT
+
+        if filter_query.pagination is not None and filter_query.pagination.limit is not None:
+            entries_limit = filter_query.pagination.limit
+
```

Comment:
this one is not needed. Check other cases but this refers to the limit that the user has regarding the amount of entries that can be displayed. This is fine being -1 for premium